pkg/prometheus/server/statefulset.go

```diff
@@ -303,6 +305,10 @@ func makeStatefulSetSpec(
 		promArgs = append(promArgs, monitoringv1.Argument{Name: "storage.tsdb.min-block-duration", Value: thanosBlockDuration})
 	}
 
+	if allowOverlappingCompaction && cpf.TSDB != nil && cpf.TSDB.OutOfOrderTimeWindow != "" && disableCompaction {
```

Comment:
I suggest that we don't make it a visible option and automatically turn on overlapping compaction when all conditions are met.
```suggestion
	if cpf.TSDB != nil && cpf.TSDB.OutOfOrderTimeWindow != "" && disableCompaction && cg.WithMinimumVersion("2.55.0").IsCompatible() {
```

Reply:
> I suggest that we don't make it a visible option and automatically turn on overlapping compaction when all conditions are met.
> 
> ```suggestion
> 
> 	if cpf.TSDB != nil && cpf.TSDB.OutOfOrderTimeWindow != "" && disableCompaction && cg.WithMinimumVersion("2.55.0").IsCompatible() {
> 
> ```

Thanks. I got it. 
But i s it possible for users to actively set it to false? 🤔

Reply:
I understand that the suggestion was to not make it a configurable option from the user: check for the 3 conditions and enable behind the scenes. (which is not what the current implementation does)
Joining @dongjiang1989 though, should we still allow the user to set it to false @simonpasquier ? It sounds reasonable, I won't be able to give a use case though.

Reply:
> The option is hidden in Prometheus (https://github.com/prometheus/prometheus/blob/e0260930d6ffbd945f7b6d030f6258b27078a442/cmd/prometheus/main.go#L408-L409) and the only (?) use case is when used with the Thanos sidecar to upload blocks to object storage. To avoid API bloat, I'd prefer not to expose it for now.

Thanks @simonpasquier . I got it.  
Remove `allowOverlappingCompaction` it done.